docs/docs/tutorial/part-7/index.mdx

```diff
@@ -0,0 +1,492 @@
+hero_image_credit_text: "Anthony Duran"
+hero_image_credit_link: "https://unsplash.com/photos/eLUBGqKGdE4"
+// highlight-end
+---
+
+...
+```
+
+```mdx:title=blog/yet-another-post/index.js
+---
+title: "Yet Another Post"
+date: "2021-07-25"
+// highlight-start
+hero_image: "./jane-almon-7rriIaBH6JY-unsplash.jpg"
+hero_image_alt: "A white pitbull wearing big googly-eye glasses"
+hero_image_credit_text: "Jane Almon"
+hero_image_credit_link: "https://unsplash.com/photos/7rriIaBH6JY"
+// highlight-end
+---
+
+...
+```
+
+
+Now that your hero images are set up, it's time to connect them to the data layer so you can pull them into your blog post page template.
+
+## Install and configure `gatsby-transformer-sharp`
+
+In order to use the `GatsbyImage` component, you'll need to add the `gatsby-transformer-sharp` transformer plugin to your site.
+
+When Gatsby adds nodes to the data layer at build time, the `gatsby-transformer-sharp` plugin looks for any File nodes that end with an image extension (like `.png` or `.jpg`) and creates an ImageSharp node for that file.
+
+![A diagram showing how ImageSharp nodes get created from File nodes in the data layer](./data-layer-with-imagesharp-nodes.png)
+
+1. In the terminal, run the following command to install `gatsby-transformer-sharp`:
+    ```shell
+    npm install gatsby-transformer-sharp
+    ```
+
+2. Add `gatsby-transformer-sharp` to the `plugins` array in your `gatsby-config.js` file.
+
+```js:title=gatsby-config.js
+module.exports = {
+  siteMetadata: {
+    title: "My First Gatsby Site",
+  },
+  plugins: [
+    // ...existing plugins
+    "gatsby-transformer-sharp", // highlight-line
+  ],
+}
+```
+
+Now that you've added `gatsby-transformer-sharp` to your site, you'll need to restart your local development server to see the changes in GraphiQL. You'll take a closer look at GraphiQL in the next step.
+
+
+## Render hero image in the blog post page template
+
+Now that you have all the necessary tools in place, it's time to add your hero image to your blog post page template.
+
+### Task: Use GraphiQL to build the query
+
+First, you'll use GraphiQL to add the hero image frontmatter fields to the GraphQL query for your blog post page template.
+
+1. Open GraphiQL by going to `localhost:8000/___graphql` in a web browser. Start by copying your existing blog post page query into the GraphiQL Query Editor pane. Run it once to make sure everything is working correctly.
+
+<Announcement style={{marginBottom: "1.5rem"}}>
+
+**Note:** You'll need to set up an object in the Query Variables pane with an `id` that matches one of your posts. Refer to [Part 6 section on query variables](/docs/tutorial/part-6/#render-post-contents-in-the-blog-post-page-template) if you need a refresher on how to set that up.
+
+</Announcement>
+
+```graphql
+query ($id: String) {
+  mdx(id: {eq: $id}) {
+    frontmatter {
+      title
+      date(formatString: "MMMM D, YYYY")
+    }
+    body
+  }
+}
+```
+
+```json
```

Comment:
If you drop the filtering by `id` the result of this might be different from what people would see